core/src/main/scala/org/bitcoins/core/bloom/BloomFilter.scala

```diff
@@ -142,16 +142,18 @@ sealed abstract class BloomFilter extends NetworkElement {
 
   /**
     * Checks if the transaction's txid, or any of the constants in it's scriptPubKeys/scriptSigs match our BloomFilter
-    * See [[https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki#filter-matching-algorithm BIP37]]
+    *
+    * @see [[https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki#filter-matching-algorithm BIP37]]
     * for exact details on what is relevant to a bloom filter and what is not relevant
     */
   def isRelevant(transaction: Transaction): Boolean = {
     val scriptPubKeys = transaction.outputs.map(_.scriptPubKey)
     //pull out all of the constants in the scriptPubKey's
     val constantsWithOutputIndex = scriptPubKeys.zipWithIndex.flatMap {
       case (scriptPubKey, index) =>
-        val constants = scriptPubKey.asm.filter(_.isInstanceOf[ScriptConstant])
-        constants.map(c => (c, index))
+        scriptPubKey.asm.collect {
+          case c: ScriptConstant => (c, index)
```

Comment:
Just a heads up, I think i have a bug here implementing this. This won't be _only_ data pushes, but will include things like `ScriptNumber`, `OP_0`,`OP_1`, etc...

Reply:
There are quite a few things that inherit from `ScriptConstant`